core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -118,6 +118,23 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] { self =>
   def char(index: Int)(implicit ev: A <:< Char): Char =
     ev(apply(index))
 
+  /**
+   * Pads to the specified number of bits.
+   *  - if n > size, chunk is 0-padded to length n.
+   *  - if n <= size, chunk is left intact.
+   *
+   * @example
+   * {{{
+   *  Chunk(true, false).padToMod(8) => Chunk(true, false, false, false, false, false, false, false)
+   * }}}
+   */
+  def padToMod(n: Int)(implicit ev: A <:< Boolean): Chunk[Boolean] =
```

Comment:
This needs to have a lower level implementation on the `BitChunk` data types to have the performance we are looking for.